### PR TITLE
Add list of selected scientific papers using signac

### PR DESCRIPTION
**Pull request by bdice**
## Description
Add a list of selected scientific papers using signac, with brief descriptions. I tried to cover several different fields of research and multiple research groups.

Preview: https://signac--142.org.readthedocs.build/en/142/scientific_papers.html

See also (for inspiration):
- https://umap-learn.readthedocs.io/en/latest/scientific_papers.html
- https://glotzerlab.engin.umich.edu/hoomd-blue/publications.html

## Motivation and Context
Shows real-world research that leverages the signac framework.

## Checklist:
- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-docs/blob/master/ContributorAgreement.md).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md#code-style) of this project.

### CODE REVIEW

**Comment by bdice** [comment]
@csadorf @vyasr @atravitz I tagged you for review since you've all expressed interest in this before.

File: docs/source/scientific_papers.rst [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+=================
+Scientific Papers
+=================
+
+The **signac** framework has been cited over 50 times, and has been used in a range of scientific fields with various types of computational workflows.
+Here, we showcase a few examples of scientific papers that have used **signac**.
+We encourage users to add their own publications to this list via a pull request.
+To find additional papers using **signac**, please refer to citations of the articles mentioned in :ref:`acknowledge`.
+
+- *Scalable Screening of Soft Matter: A Case Study of Mixtures of Ionic Liquids and Organic Solvents* :cite:`Thompson2019`
```
Comment (csadorf) [inline docs/source/scientific_papers.rst:12]: This is a nice selection, but I think the formatting makes it actually very hard to parse. Maybe we can use something like [panels](https://sphinx-panels.readthedocs.io/) or some other visual divider to make the individual papers more distinguishable. The UMAP page has graphics for each paper, however I understand that it might be difficult to obtain permissions to represent abstract graphics for all listed papers.

I would also suggest to use a slightly more elaborate reference style with at least the journal and year or so. Something in between the representation of UMAP and HOOMD-blue. My assumption is that anyone who makes their way to that page won't be intimidated by the standard scientific reference format.

**Review by csadorf — COMMENTED** [review]
I've made some suggestions for improved representation. If you could apply those first then I can have a look at the actual content.

File: docs/source/scientific_papers.rst [hunk 2]
```diff
@@ -0,0 +1,42 @@
+.. _scientific-papers:
+
+=================
+Scientific Papers
+=================
+
+The **signac** framework has been cited over 50 times, and has been used in a range of scientific fields with various types of computational workflows.
+Here, we showcase a few examples of scientific papers that have used **signac**.
+We encourage users to add their own publications to this list via a pull request.
+To find additional papers using **signac**, please refer to citations of the articles mentioned in :ref:`acknowledge`.
```
[the remark this answers is not in the document]
Comment (bdice) [reply docs/source/scientific_papers.rst:10]:
```suggestion
To find additional papers using **signac**, please refer to citations of the articles mentioned in :ref:`acknowledge` or `citing works on Google Scholar <https://scholar.google.com/scholar?cites=14270565839129827405,11696507121891896813>`__.
```

File: docs/source/scientific_papers.rst (hunk 1 above)
Comment (bdice) [reply docs/source/scientific_papers.rst]: @csadorf I changed the style pretty significantly. See if this looks better to you.

Comment (csadorf) [reply docs/source/scientific_papers.rst]: Yes, that's much easier to parse. Now, the next question is, what is the order based on? Would anti-chronological make sense?

Comment (bdice) [reply docs/source/scientific_papers.rst]: Sure, I can do newest to oldest if that’s better.

Comment (csadorf) [reply docs/source/scientific_papers.rst]: Yes, I think that would be best, but the previous order was not chronological in any way, which is why I was wondering what criterion you used.

Comment (bdice) [reply docs/source/scientific_papers.rst]: I updated the papers to be reverse-chronological. I didn't really have an order in mind before, the order was just copied from the listings in the 2021 SciPy Proceedings paper.

File: docs/source/scientific_papers.rst [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+=================
+Scientific Papers
+=================
+
+The **signac** framework has been cited over 50 times, and has been used in a range of research fields with various types of computational workflows.
+Here, we showcase a few examples of scientific papers that have used **signac**.
+To find additional papers using **signac**, please refer to citations of the articles mentioned in :ref:`acknowledge` or this list of `citing works on Google Scholar <https://scholar.google.com/scholar?cites=14270565839129827405,11696507121891896813>`__.
+
+We encourage users to add publications citing **signac** to this list via a pull request.
+To add a paper, put citation information in ``signac.bib`` and follow the formatting of this page.
```
Comment (csadorf) [inline docs/source/scientific_papers.rst:12]:
```suggestion
To add a paper, put citation information in `signac.bib <https://github.com/glotzerlab/signac-docs/edit/master/docs/source/signac.bib>`__ and follow the formatting of this page.
```
I've actually had a little bit of trouble finding that file. So this might make it easier.

**Review by csadorf — APPROVED** [review]
One more suggestion, otherwise ready!

File: docs/source/scientific_papers.rst [hunk 4]
```diff
@@ -0,0 +1,85 @@
+.. _scientific-papers:
+
+=================
+Scientific Papers
+=================
+
+The **signac** framework has been cited over 50 times, and has been used in a range of research fields with various types of computational workflows.
```
Comment (cbkerr) [inline docs/source/scientific_papers.rst:7]:
```suggestion
The **signac** framework has been cited over 50 times and has been used in a range of research fields with various types of computational workflows.
```

**Review by cbkerr — APPROVED** [review]
I like putting the menu item in the "community" section.

Information for using it in the wild seems like something that could _also_ be nice to have on the main signac.io page to capture more interest.
